server/services/v1/database/secondary_indexer.go

```diff
@@ -440,20 +443,26 @@ func (q *SecondaryIndexerImpl) buildTableRows(tableData *internal.TableData) ([]
 
 func (q *SecondaryIndexerImpl) buildTSRows(tableData *internal.TableData) ([]IndexRow, error) {
 	timeStamps := []struct {
-		ts    *internal.Timestamp
-		field schema.ReservedField
+		ts          *internal.Timestamp
+		field       schema.ReservedField
+		shouldIndex bool
 	}{
 		{
 			tableData.CreatedAt,
 			schema.CreatedAt,
+			true,
 		},
 		{
 			tableData.UpdatedAt,
 			schema.UpdatedAt,
+			true,
 		},
 	}
 
 	rows := make([]IndexRow, 0, len(timeStamps))
+	if !q.coll.SecondaryIndexMetadata() {
```

Comment:
Can we assume now that Replace API doesn't need to read anything if there is no index defined by the user and the index metadata is false?